keyboards/ergodox_ez/keymaps/dctucker/keymap.c

```diff
@@ -0,0 +1,344 @@
+	),
+	[3] = LAYOUT_ergodox_pretty(
+		_______, KC_F1  , KC_F2  , KC_F3  , KC_F4  , KC_F5  , _______,  KC_PIPE, KC_CIRC, KC_AMPR, KC_PAST, KC_LPRN, KC_RPRN, _______,
+		_______, _______, _______, _______, _______, _______, _______,  KC_PIPE, KC_EXLM, KC_AT  , KC_HASH, KC_DLR , KC_PERC, KC_BSLS,
+		KC_ESC , _______, _______, _______, _______, _______,                    KC_LEFT, KC_DOWN, KC_UP  , KC_RGHT, KC_COLN, _______,
+		_______, _______, _______, _______, _______, _______, _______,  _______, _______, BACKN  , EARROW , ARROW  , COMMENT, _______,
+		_______, _______, _______, _______, _______,                                      _______, _______, _______, KC_LPRN, KC_RPRN,
+		                                             _______, _______,  _______, _______,
+		                                                      _______,  _______,
+		                                    _______, _______, _______,  _______, KC_PENT, _______
+	),
+	[4] = LAYOUT_ergodox_pretty(
+		_______, _______, _______, _______, _______, _______, _______,  _______, _______, _______, _______, _______, _______, _______,
+		_______, _______, _______, _______, _______, _______, _______,  _______, _______, _______, _______, _______, _______, _______,
+		KC_CAPS, _______, _______, _______, _______, _______,                    _______, _______, _______, _______, _______, _______,
+		_______, _______, _______, _______, _______, _______, _______,  _______, _______, _______, _______, _______, _______, _______,
+		_______, _______, _______, _______, _______,                                      _______, _______, _______, _______, _______,
+		                                             _______, _______,  _______, _______,
+		                                                      _______,  _______,
+		                                    _______, _______, _______,  _______, _______, _______
+	),
+};
+
+const uint16_t PROGMEM fn_actions[] = {
+  [1] = ACTION_LAYER_TAP_TOGGLE(1)
+};
+
+bool xtra_sent = true;
+bool ctl_xtra_sent = true;
+uint16_t lctl_timer, rctl_timer;
+bool caps = false;
+
+void keyboard_post_init_user(void) {
+	led_set_user(host_keyboard_leds());
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	switch (keycode) {
+		// dynamically generate these.
+		case EPRM:
+			if (record->event.pressed) {
+				eeconfig_init();
+			}
+			return false;
+			break;
+		case RGB_SLD:
+			if (record->event.pressed) {
+				rgblight_mode(1);
+			}
+			return false;
+			break;
+		case CTLESC:
+			if( record->event.pressed ){
+				ctl_xtra_sent = false;
+				lctl_timer = timer_read();
+				register_code(KC_LCTL);
+			} else {
+				unregister_code(KC_LCTL);
+				if( ! ctl_xtra_sent ){
+					if( timer_elapsed(lctl_timer) < 500 ){
+						if( host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK) ){
+							add_key(KC_RSFT);
+							add_key(KC_ESC);
+							send_keyboard_report();
+							del_key(KC_ESC);
+							del_key(KC_RSFT);
+							send_keyboard_report();
```

Comment:
```suggestion
							tap_code16(RSFT(KC_ESC));
```

Reply:
Interestingly this change increases the firmware size by 268 bytes.